mitodl/mit-learn
# similar resources carousel

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/5878

### Description (What does it do?)
This PR adds two `ResourceCarousel`s to the new learning resource drawer based on the new "similar resources" APi endpoints. Both the opensearch and vector based results are displayed below the resource information.

### Screenshots (if appropriate):
![image](https://github.com/user-attachments/assets/eedb3d35-ab5b-4493-9043-2045e91d9bea)
![image](https://github.com/user-attachments/assets/a2c1295f-bf18-49ce-9b4b-e71cfc631186)

### How can this be tested?
 - If you have Posthog set up locally, enable the `lr_drawer_v2` flag
 - If you don't have Posthog set up locally, you can force `drawerV2` to be `true` in `LearningResourceDrawer.tsx`
 - Spin up this branch of `mit-learn`
 - Generate vector embeddings by running `docker compose exec web python manage.py generate_embeddings --skip-contentfiles --recreate-collections --all`
 - Visit the search page at http://localhost:8062/search
 - View a variety of learning resources and ensure that the related resources carousels appear
 - Click the cards in the carousels to navigate to other learning resources and ensure that the drawer reloads with the new resource seamlessly
